src/Streamly/Internal/Data/Stream/StreamD/Transform.hs

```diff
@@ -1203,29 +1204,42 @@ slicesBy p (Stream step1 state1) = Stream step (Just (state1, 0, 0))
 data RollingMapState s a = RollingMapInit s | RollingMapGo s a
 
 {-# INLINE rollingMapM #-}
-rollingMapM :: Monad m => (a -> a -> m b) -> Stream m a -> Stream m b
+rollingMapM :: Monad m => (Maybe a -> a -> m b) -> Stream m a -> Stream m b
 rollingMapM f (Stream step1 state1) = Stream step (RollingMapInit state1)
+
```

Comment:
Just use `RollingMapGo state1 Nothing` here. The RollingMapInit state can be removed.